operator/v1/types_clustermanager.go

```diff
@@ -232,19 +232,62 @@ const (
 	FeatureGateModeTypeDisable FeatureGateModeType = "Disable"
 )
 
+// DefaultClusterManagerConfiguration represents customized configurations for clustermanager in the Default mode
+type DefaultClusterManagerConfiguration struct {
+	// RegistrationWebhookConfiguration represents the customized webhook-server configuration of registration.
+	// +optional
+	RegistrationWebhookConfiguration DefaultWebhookConfiguration `json:"registrationWebhookConfiguration,omitempty"`
+
+	// WorkWebhookConfiguration represents the customized webhook-server configuration of work.
+	// +optional
+	WorkWebhookConfiguration DefaultWebhookConfiguration `json:"workWebhookConfiguration,omitempty"`
+}
+
 // HostedClusterManagerConfiguration represents customized configurations we need to set for clustermanager in the Hosted mode.
 type HostedClusterManagerConfiguration struct {
 	// RegistrationWebhookConfiguration represents the customized webhook-server configuration of registration.
 	// +optional
-	RegistrationWebhookConfiguration WebhookConfiguration `json:"registrationWebhookConfiguration,omitempty"`
+	RegistrationWebhookConfiguration HostedWebhookConfiguration `json:"registrationWebhookConfiguration,omitempty"`
 
 	// WorkWebhookConfiguration represents the customized webhook-server configuration of work.
 	// +optional
-	WorkWebhookConfiguration WebhookConfiguration `json:"workWebhookConfiguration,omitempty"`
+	WorkWebhookConfiguration HostedWebhookConfiguration `json:"workWebhookConfiguration,omitempty"`
 }
 
-// WebhookConfiguration has two properties: Address and Port.
+// WebhookConfiguration represents customization of webhook servers
```

Comment:
actually it is not WebhookConfiguration, I think we could also use this for controller to set healthiness and metrics endpoint?

Reply:
True. Maybe `BindConfiguration`?

Realized there is an issue for Hosted mode. The Port field is only the external endpoint for k8s API to reach the validating webhook, but if you set HostNetwork then you would also want to control the internal port bind to avoid port collision on the host net.

Perhaps a better API would be something like this:
```yaml
spec:
  ...
  deployOption:
    hosted:
      workWebhookConfiguration:
        port: 443
        address: work.webhook.example.com
        bindConfig:
          port: 19443
          healthProbeBindAddress: ":18000"
          metricsBindAddress: ":18080"
          hostNetwork: true
      registrationWebhookConfiguration:
        ...
    mode: Hosted
```

Reply:
It looks like controller servers work very differently from the webhooks. They use the `openshift/library-go` controller builder, which only takes a single listen address:port. Whereas the webhooks use `controller-runtime` which has separate binds for each. So I don't think it makes sense to re-use this configuration for controllers (unless they get migrated to controller-runtime in the future)

Reply:
Well it's not just that it can bind separately, they are separate listeners. No option to share port with the webhook server. As a result, when using hostNetwork mode, it is important to have control over the binds.

Reply:
I understand the port should be different, I am wondering if it is necessary to bind different HOST/IP

Reply:
I see. I figured if you can bind hostNet then it is reasonable to allow fully configuring the bind for metrics/healthz. Not strictly necessary though (I had no intention of using it, I'm just binding ":<port>" for all of these).